libs/core/src/main/java/org/elasticsearch/common/compatibility/CompatibleVersion.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.compatibility;
+
+/**
+ * A enum representing versions which are used by a REST Compatible API.
+ * A CURRENT instance, represents a major Version.CURRENT from server module.
+ *
+ * Only major versions are supported.
+ */
+public enum CompatibleVersion {
+
+    V_8(8),
+    V_7(7),
+    V_6(6);//used in testing, to prove validation
+
+    // This needs to be aligned with Version.CURRENT
+    public static final CompatibleVersion CURRENT = V_8;
```

Comment:
It would be great if we could hide the CURRENT, so a user can't type `CompatibleVersion.CURRENT` since in all but the core cases (like parsing the header) it would be incorrect to do so. 

It's a pretty minor concern ... maybe a simple comment would suffice ? , or maybe a getCurrentVersion() (with java doc saying dont use this) instead to avoid it looking like one of the enum options?

Reply:
good point, I feel `getCurrentVersion` can help here.